tests/test_slurm.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from IPython.testing.globalipapp import start_ipython
+
+# pylint: disable=redefined-outer-name
+
+
+@pytest.fixture(scope="session")
+def session_ip():
+    """Get an IPython shell"""
+    yield start_ipython()
+
+
+@pytest.fixture(scope="function")
+def ip(session_ip):
+    """Prepare IPython shell for running unit tests"""
+    session_ip.run_cell(raw_cell="import qililab.slurm")
+    yield session_ip
+
+
+def test_submit_job(ip):
+    ip.run_cell(raw_cell="a=1\nb=1")
+    ip.run_cell_magic(
+        magic_name="submit_job",
+        line="-o results -d debug -l slurm_job_data_tests -n unit_test -e local",
+        cell="results = a+b ",
+    )
+    assert ip.user_global_ns["results"].result() == 2
```

Comment:
Are these tests creating actual files? If so, we should make sure to delete them at the end of every test!

We can do this with [teardown methods](https://stackoverflow.com/questions/26405380/how-do-i-correctly-setup-and-teardown-for-my-pytest-class-with-tests)

Reply:
yes, this seems quite powerful https://docs.pytest.org/en/latest/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization. But I don't think there's need to implement such a thing for this tests. Thus, I implemented a manual teardown :)